### PR TITLE
[BD-24][TNL-7607] BB-3072: Deep Linking response endpoint and model implementation 

### PR DESCRIPTION
**Pull request by giovannicimolin**
This PR implements the Deep Linking response endpoint, which tools use to send back information to the LMS after the user selects the content and configuration in the Tool's UI. 

Split from #112 

**Testing instructions:**
__Incomplete, tool configuration instructions TBD__
1. Checkout this branch.
2. Set up a local LTI tool (recommended: https://github.com/dmitry-viskov/pylti1.3-django-example) and ngrok tunnel.
3. Apply this patch to allow Deep Link launches:
```
diff --git a/lti_consumer/lti_xblock.py b/lti_consumer/lti_xblock.py
index df0cf16..3e5cadd 100644
--- a/lti_consumer/lti_xblock.py
+++ b/lti_consumer/lti_xblock.py
@@ -1378,7 +1378,7 @@ class LtiConsumerXBlock(StudioEditableXBlockMixin, XBlock):
             # Retrieve and set LTI 1.3 Launch start URL
             lti_block_launch_handler = get_lti_1p3_launch_start_url(
                 block=self,
-                deep_link_launch=False,
+                deep_link_launch=True,
                 hint=str(self.location)  # pylint: disable=no-member
             )
```
4. Do a LTI launch (from the LMS).
5. Select an option in the LTI tool's UI.
6. Go to http://localhost:18000/admin/lti_consumer/ltidlcontentitem/ and check that the option you selected is stored in the model. Make note of the ID.
7. Do the launch again, and check that the old option was deleted and only the new one is stored.

**Author notes:**
This does not implement content presentation, just stores the data the user selected in the models.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @giovannicimolin! I've created [BLENDED-681](https://openedx.atlassian.net/browse/BLENDED-681) to keep track of it in Jira.   More details are on the [BD-24](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1545076784/BD-24+LTI+v1.3+Improvements+Advantage+Support) project page.


When this pull request is ready, tag your edX technical lead.

**Review by shimulch — APPROVED** [review]
LGTM :+1: 

- [x] I tested this in my local devstack. I've used the python LTI tool mentioned in the test instruction. LTI launch works correctly. And, clicking a DL link from the LTI Tool correctly redirects me to the DL response endpoint on LMS. I've checked, the DL response content is saved and gets replaced by a new one when performing the DL flow again.
- [x] I read through the code
- [N/A] I checked for accessibility issues
- [x] Includes documentation
- [N/A] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

**Comment by giovannicimolin** [comment]
@nedbat This is ready for review again. I've fixed the tests and added one extra test case to cover a few missing lines.

About logging, can I implement it in a separate follow-up PR? This is already pretty big.

File: lti_consumer/lti_1p3/consumer.py [hunk 1]
```diff
@@ -493,3 +515,101 @@ def enable_ags(
...
+        if not message_type == "LtiDeepLinkingResponse":
+            raise exceptions.InvalidClaimValue("Token isn't a Deep Linking Response message.")
+
+        # Check if supported contentitems were returned
+        content_items = deep_link_response.get(
+            'https://purl.imsglobal.org/spec/lti-dl/claim/content_items',
+            # If not found, return empty list
+            [],
+        )
+        if any([
```
Comment (nedbat) [inline lti_consumer/lti_1p3/consumer.py:608]: Python nit: no need for square brackets here.  any() will work with a generator expression.

File: lti_consumer/lti_1p3/deep_linking.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        # Consctruct Deep Linking Claim
+        deep_linking_claim = {
+            "accept_types": accept_types_claim,
+            "accept_presentation_document_targets": [
+                "iframe",
+                "window",
+                "embed"
+            ],
+            # Only accept a single item return from Deep Linking operation.
+            "accept_multiple": True,
```
Comment (nedbat) [inline lti_consumer/lti_1p3/deep_linking.py:56]: This line doesn't match the comment before it.  Which is right?

File: lti_consumer/lti_1p3/deep_linking.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            # Automatically saves Content Items without asking to user
+            "auto_create": True,
+            # Other parameters
+            "title": title,
+            "text": description,
+            "deep_link_return_url": self.deep_linking_return_url
```
Comment (nedbat) [inline lti_consumer/lti_1p3/deep_linking.py:62]: You'll be glad later if you include trailing commas now :)

File: lti_consumer/lti_1p3/consumer.py (hunk 1 above)
Comment (giovannicimolin) [reply lti_consumer/lti_1p3/consumer.py]: Fixed!

File: lti_consumer/lti_1p3/deep_linking.py (hunk 2 above)
Comment (giovannicimolin) [reply lti_consumer/lti_1p3/deep_linking.py]: You're right, thanks for catching the issue.

File: lti_consumer/lti_1p3/deep_linking.py (hunk 3 above)
Comment (giovannicimolin) [reply lti_consumer/lti_1p3/deep_linking.py]: Fixed. Is there a way we configure `pylint` or `pycodestyle` to catch these in the future?

**Comment by nedbat** [comment]
I've tested this with the IMS Validator. It passes both the Core and the Deep Linking sections.

**Pull request merged**